boa_engine/src/bytecompiler/function.rs

```diff
@@ -189,8 +197,77 @@ impl FunctionCompiler {
             compiler.emit_opcode(Opcode::Yield);
         }
 
-        compiler.create_script_decls(body, false);
-        compiler.compile_statement_list(body, false, false);
+        let var_names = top_level_var_declared_names(body);
```

Comment:
https://tc39.es/ecma262/#sec-functiondeclarationinstantiation

Reply:
Also, should we extract this into a function? Since this is also used in `ClassElement::StaticBlock` and in the `constructor` function of a class.